src/Microsoft.OpenApi/Services/OpenApiSecurityRequirementComparer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Services
+{
+    /// <summary>
+    /// Defines behavior for comparing properties of <see cref="OpenApiSecurityRequirement"/>.
+    /// </summary>
+    public class OpenApiSecurityRequirementComparer : OpenApiComparerBase<OpenApiSecurityRequirement>
+    {
+        /// <summary>
+        /// Executes comparision against source and target <see cref="OpenApiSecurityRequirement"/>.
+        /// </summary>
+        /// <param name="sourceSecurityRequirement">The source.</param>
+        /// <param name="targetSecurityRequirement">The target.</param>
+        /// <param name="comparisonContext">Context under which to compare the source and target.</param>
+        public override void Compare(
+            OpenApiSecurityRequirement sourceSecurityRequirement,
+            OpenApiSecurityRequirement targetSecurityRequirement,
+            ComparisonContext comparisonContext)
+        {
+            if (sourceSecurityRequirement == null && targetSecurityRequirement == null)
+            {
+                return;
+            }
+
+            if (sourceSecurityRequirement == null || targetSecurityRequirement == null)
+            {
+                comparisonContext.AddOpenApiDifference(
+                    new OpenApiDifference
+                    {
+                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Update,
+                        SourceValue = sourceSecurityRequirement,
+                        TargetValue = targetSecurityRequirement,
+                        OpenApiComparedElementType = typeof(OpenApiSecurityRequirement),
+                        Pointer = comparisonContext.PathString
+                    });
+
+                return;
+            }
+
+            var newSecuritySchemesInTarget = targetSecurityRequirement.Keys
+                .Where(targetReq => sourceSecurityRequirement.Keys.All(
+                    sourceReq => sourceReq.Reference.Id != targetReq.Reference.Id)).ToList();
+
+            foreach (var newSecuritySchemeInTarget in newSecuritySchemesInTarget)
+            {
+                WalkAndAddOpenApiDifference(
+                    comparisonContext,
+                    newSecuritySchemeInTarget.Reference.Id,
```

Comment:
> newSecuritySchemeInTarget.Reference.Id, [](start = 19, length = 40)

Does this yield the right string (i.e. just the security scheme name, not the json pointer)?

Reply:
Yes i have test for it in OpenApiComparerTests

---
In reply to: [223516346](https://github.com/Microsoft/OpenAPI.NET/pull/328#discussion_r223516346) [](ancestors = 223516346)